torch/csrc/jit/frontend/schema_matching.cpp

```diff
@@ -616,12 +631,48 @@ Value* emitBuiltinCall(
   const auto& variants = getAllOperatorsFor(name);
   const auto& builtin_functions = getAllBuiltinFunctionsFor(name);
 
+  // first let's set the graph's version
+  auto graph_version = graph.get_op_version();
+
   std::stringstream failure_messages;
   std::vector<const FunctionSchema*> schemas;
+  // we append them later to schemas because
+  // parseSchema returns rvalue which can not
+  // be casted to const pointer.
+  std::vector<FunctionSchema> upgrader_schemas;
   schemas.reserve(variants.size());
   for (const std::shared_ptr<Operator>& op : variants) {
-    schemas.push_back(&op->schema());
+    bool found_upgrader = false;
+    auto op_overload_name = op->schema().overload_name();
+    auto op_name = op->schema().operator_name().name +
+        ((op_overload_name != "") ? "." + op_overload_name : "");
+    if (graph_version.has_value()) {
+      auto version_entry = get_operator_version_map().find(op_name);
+      if (version_entry != get_operator_version_map().end()) {
+        auto old_schema_entry =
+            findUpgrader(version_entry->second, graph_version.value());
+        if (old_schema_entry.has_value()) {
+          FunctionSchema old_schema =
+              parseSchema(old_schema_entry.value().old_schema);
+          upgrader_schemas.push_back(old_schema);
+          found_upgrader = true;
+        } else {
+          if (!isOpCurrentBasedOnUpgraderEntries(
+                  version_entry->second, graph_version.value())) {
+            TORCH_INTERNAL_ASSERT(false, "Valid upgrader must be present");
```

Comment:
Is this also about deprecated upgrader? If so, the error message can be improved. Current error message doesn't explain what happened and isn't actionable.

Besides, `torch_internal_assert` implies that something is wrong with TorchScript implementation (ie some invariant that should never be broken is broken), can we throw a normal exception here instead?